### DISCUSSION TITLE
Render lazy property.

### DISCUSSION THREAD

**Discussion by AsadCSE** [harttle/liquidjs#676] [Q&A] [answered]
Yesterday while working i found that with shopify liquid it is possible to render a property inside an object on demand which may require additional db operation. Like the "collection" object in shopify liquid does not have a properly called "products" but when used like {{collection.products}} it outputs product list and instead of [productDrop] it shows [lazyProductDrop]. How can i achieve such usage in liquidjs? Please provide me an example.

Thank you.

**Comment by harttle** [comment, marked as answer]
Hi @AsadCSE , please try Drops: https://liquidjs.com/tutorials/drops.html

And alternatively, data properties in LiquidJS can be promises, like:

```javascript
liquid.parseAndRenderAsync("{{foo}}", {
    foo: fetchSomethingAsync().then(obj => obj.foo)
})
```

**Reply by AsadCSE** [reply]
Thank you @harttle for your quick reply.

I have tried the toLiquid method and that works like a charm.
